app/assets/javascripts/discourse/app/models/category.js

```diff
@@ -386,6 +387,28 @@ export default class Category extends RestModel {
     return data.sortBy("read_restricted");
   }
 
+  static async asyncHierarchicalSearch(term, opts) {
```

Comment:
Will this endpoint only be used if there is a search term? Meaning, for the initial load of the modal, the current endpoint works OK?